### DISCUSSION TITLE
ORAS Support for labels

### DISCUSSION THREAD

**Discussion by KevinDMack** [oras-project/oras#686] [Q&A] [answered]
So we have a process by which our build process is building a artifact using oras, and it works great for building the artifact, but our end delivery requires that we have a label on the resulting artifact / image.  Is there a way to do this when building before the push?  Similar to how we would identify the label in the Dockerfile.

**Comment by qweeah** [comment, marked as answer]
Hey Kevin, thanks for initiating this discussion. I am not sure if you are trying to add annotations to the pushed manifest. If so, you can try below command:
```
Example - Push repository with manifest annotations
  oras push --annotation "key1=val1"  --annotation "key2=val2" localhost:5000/hello:latest hi.txt
```
It works similarly for attaching:
```
Example - Attach file 'hi.txt' and add manifest annotations
  oras attach --artifact-type doc/example --annotation "key=val" localhost:5000/hello:latest hi.txt
```

**Comment by rpardini** [comment]
Hmm, I found this looking for a way to put labels on the pushed artifacts.
As the OP said, I'd like to add labels to the "main" manifest (sorry for terminology), like `LABEL ` would do in a Dockerfile.
ORAS cli does create a label already, but I can't find a way to change it or add others.
```json
  "labels": {
    "org.opencontainers.image.created": "2023-02-03T14:37:58Z"
  }
```

**Reply by rpardini** [reply]
Scratch that. If I use `"--annotation" "org.opencontainers.image.description=some description"` it magically becomes a label! On `0.16.0` at least...
